### PR TITLE
added code to delete overlay images at slave mount point

### PR DESCRIPTION
**Pull request by sabose**
and to wait for snapshot deletion.

Signed-off-by: Sahina Bose sabose@redhat.com

### CODE REVIEW

File: vmbackup.py [hunk 1]
```diff
@@ -185,12 +279,21 @@ def connect(url, username, password):
...
+
+def configure_logging():
     logger = logging.getLogger()
-    handler = logging.StreamHandler()
-    formatter = logging.Formatter('%(asctime)s %(name)-12s %(levelname)-8s %(message)s')
-    handler.setFormatter(formatter)
-    logger.addHandler(handler)
-    logger.setLevel(logging.DEBUG)
+    try:
+        fileConfig("ovirt_georep_backup_log.conf")
```
Comment (aravindavk) [inline vmbackup.py]: will it look for conf in current dir? or absolute path required?

File: vmbackup.py [hunk 2]
```diff
@@ -85,6 +127,35 @@ def parse_input():
...
 
+
+def wait_for_snapshot_deletion(vm, snapshotid):
+    while True:
+        snapshot = vm.snapshots.get(id=snapshotid)
+        if snapshot is not None:
+            logger.debug ("Snapshot status: " + snapshot.get_snapshot_status())
+            time.sleep(10)
+        else:
+            logger.error ("Snapshot deleted for VM: " + vm.name)
```
Comment (aravindavk) [inline vmbackup.py]: Not error. Info?

File: vmbackup.py [hunk 3]
```diff
@@ -151,16 +233,27 @@ def main(args):
...
             vm = vm_to_commit['vm']
-            logger.info("Delete snapshot for: " + vm.name)
+            logger.debug("Deleting snapshot for: " + vm.name)
             snapshot = vm_to_commit['snapshot']
-
-            # vm.commit_snapshot() - not working  "Cannot revert to Snapshot. VM's Snapshot does not exist."
+            # live merge
             snapshot.delete()
+            # wait for snapshot deletion to complete
+            wait_for_snapshot_deletion(vm, snapshot.get_id())
```
Comment (aravindavk) [inline vmbackup.py]: corner case: if snapshot delete is hung or delete failed

File: vmbackup.py (hunk 1 above)
Comment (sabose) [inline vmbackup.py]: Will make it customizable via the conf file. By default, it will look for this conf in current dir

File: vmbackup.py (hunk 2 above)
Comment (sabose) [inline vmbackup.py]: Yes, thanks!

File: vmbackup.py (hunk 3 above)
Comment (sabose) [inline vmbackup.py]: will add a timeout check for wait

**Pull request merged**